### PR TITLE
[KOGITO-2038] Setup release pipeline

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2038

Depends on
- https://github.com/kiegroup/kogito-runtimes/pull/570
- https://github.com/kiegroup/kogito-images/pull/183
- https://github.com/kiegroup/kogito-cloud-operator/pull/420

### CODE REVIEW

File: Jenkinsfile.release [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+                            deleteDir()
+                            checkout(githubscm.resolveRepository(repo, env.GIT_AUTHOR, "master", false))
+                            sh "git fetch origin"
+                            String branchRemoteResult = sh(script: "git ls-remote origin ${getReleaseBranch()} | wc -l", returnStdout: true).trim()
+                            if(Integer.parseInt(branchRemoteResult) > 0){
+                                echo "Release branch already exist ... will not create it"
+                            } else {
+                                echo "Release branch ${getReleaseBranch()} does not exist ... gonna create it"
+                                githubscm.createBranch(getReleaseBranch())
+                                githubscm.pushObject("origin", getReleaseBranch(), env.AUTHOR_CREDS_ID)
```
Comment (Kevin-Mok) [inline Jenkinsfile.release]: Should this be refactored into a helper function (e.g. `createReleaseBranch`)?

**Review by Kevin-Mok — APPROVED** [review]

**Review by Ginxo — APPROVED** [review]
I don't see any technical problem here. 
Just small thing, to distinguish between `'` (String) and `"` (GString) in the whole pipeline, for instance: `IMAGE_REGISTRY = "quay.io"` should be `IMAGE_REGISTRY = 'quay.io'` or `job.getResult() == "SUCCESS"` by `job.getResult() == 'SUCCESS'`
Not important but I would suggest to change it at some point.

I can't tell about functionality

**Review by Kaitou786 — APPROVED** [review]

File: Jenkinsfile.release (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile.release]: It will be extracted into a separate job: https://issues.redhat.com/browse/KOGITO-2908

**Comment by radtriste** [comment]
@Ginxo updated with your remarks. gonna do the same on other pipelines.
For the functionality, this has been tested on Kogito 0.13.0 and adjusted. will be used for a new release on 0.13.1 next week.

**Comment by Ginxo** [comment]
@radtriste looks great!! :heart_eyes:  Can I merge?

**Comment by radtriste** [comment]
I would rather wait for the other pipelines (runtimes/images/operator) to be merged first as some parameters sent by this pipeline are not yet existing on others...
I will let you know once it should be merged.

**Pull request merged**